content/influxdb/v2.0/reference/internals/shards.md

```diff
@@ -0,0 +1,148 @@
+
+### Shard writes
+InfluxDB writes time series data to un-compacted or "hot" shards.
+When a shard is no longer actively written to, InfluxDB progressively compresses
+or [compacts](#shard-compaction) the shards.
+Compacted shards are considered "cold" shards.
+
+Typically, InfluxDB writes data to the most recent shard group, but when backfilling
+historical data InfluxDB writes to older shards that must first be un-compacted.
+When the backfill is complete, InfluxDB re-compacts the older shards.
+
+### Shard compaction
+InfluxDB compacts shards at regular intervals to compress time series data and optimize disk usage.
+InfluxDB uses the following four compaction levels:
+
+- **Level 1 (L1):** InfluxDB flushes all newly written data held in an in-memory cache to disk.
+- **Level 2 (L2):** InfluxDB compacts up to eight L1-compacted files into one or more L2 files by
+     combining multiple blocks containing the same series into fewer blocks in one or more new files.
+- **Level 3 (L3):** InfluxDB iterates over L2-compacted file blocks (over a certain size)
+  and combines multiple blocks containing the same series into one block in a new file.
+- **Level 4 (L4):** **Full compaction**—InfluxDB iterates over L3-compacted file blocks
+  and combines multiple blocks containing the same series into one block in a new file.
+
+##### Shard compaction-related configuration settings
+- [`storage-compact-full-write-cold-duration`](/influxdb/v2.0/reference/config-options/#storage-compact-full-write-cold-duration)
+- [`storage-compact-throughput-burst`](/influxdb/v2.0/reference/config-options/#storage-compact-throughput-burst)
+- [`storage-max-concurrent-compactions`](/influxdb/v2.0/reference/config-options/#storage-max-concurrent-compactions)
+- [`storage-max-index-log-file-size`](/influxdb/v2.0/reference/config-options/#storage-max-index-log-file-size)
+- [`storage-series-file-max-concurrent-snapshot-compactions`](/influxdb/v2.0/reference/config-options/#storage-series-file-max-concurrent-snapshot-compactions)
+- [`storage-series-file-max-concurrent-snapshot-compactions`](/influxdb/v2.0/reference/config-options/#storage-series-file-max-concurrent-snapshot-compactions)
+
+## Shard deletion
+The InfluxDB **retention enforcement** service routinely checks for shard groups
+older than their bucket's retention period.
+Once the start time of a shard group is beyond the bucket's retention period,
+InfluxDB deletes the shard group and associated shards and TSM files.
+
+In buckets with an infinite retention period, shards remain on disk indefinitely.
+
+{{% note %}}
+#### InfluxDB only deletes cold shards
+InfluxDB only deletes **cold** shards.
+If backfilling data beyond a bucket's retention period, the backfilled data will
+remain on disk until:
+
+1. The shard returns to a cold state.
+2. The retention enforcement service deletes the shard group.
```

Comment:
```suggestion
- The shard returns to a cold state.
- The retention enforcement service deletes the shard group.
```

Reply:
These are sequential. The shard first has to go cold. InfluxDB won't delete a hot shard.